src/app/core/public/api/impl/CircuitState.ts

```diff
@@ -2,13 +2,22 @@ import {CircuitInternal}   from "core/internal";
 import {SelectionsManager} from "core/internal/impl/SelectionsManager";
 import {CircuitView}       from "core/internal/view/CircuitView";
 
+import {Circuit}   from "../Circuit";
+import {Component} from "../Component";
+import {Port}      from "../Port";
+import {Wire}      from "../Wire";
+
 
 // Utility object to pass around the Circuit API state to the API Component/Port/etc
-export interface CircuitState {
+export interface CircuitState extends Circuit {
```

Comment:
```typescript
export interface CircuitState<
  ComponentT extends Component = Component,
  WireT extends Wire = Wire,
  PortT extends Port = Port
> extends Circuit { 
  // ...
  constructComponent(id: string): ComponentT;
  // ...
}
```

Would something like this make
```typescript
interface DigitalCircuitState extends CircuitState<DigitalComponent, DigitalWire, DigitalPort>
```
possible and remove the dup'd fields/functions in that interface?  Would this make `CircuitImpl` a nightmare?

Reply:
I like this idea, let me give it a try